File: patch_labeling_web/add_center_rectangle.m
function add_center_rectangle()
% adds a green rectangle in the image center, to indicate the patch
% the pathologist to label for.

% x and y point of the top-left corner of the rectangle
s = 200;
% side length of the rectangle
l = 100;

files = dir('images/*.png');
for file = files'
    fn = file.name;
    im = imread(['images/', fn]);
    im(s:s+l, s, 2) = 255;
    im(s, s:s+l, 2) = 255;
    im(s:s+l, s+l, 2) = 255;
    im(s+l, s:s+l, 2) = 255;
    imwrite(im, ['images/', fn]);
end

File: patch_labeling_web/add_center_rectangle.py
import cv2
import os


s = 200
l = 100

files = os.listdir('images/')
for file in files:
    if '.png' not in file: continue
    fn = 'images/' + file

    im = cv2.imread(fn)
    im[s:s + l, s:s+3, 2] = 255
    im[s:s+3, s: s + l, 2] = 255
    im[s: s + l, s + l: s + l + 3, 2] = 255
    im[s + l: s + l + 3, s: s + l, 2] = 255

    im[s:s + l, s:s+3, 0:2] = 0
    im[s:s+3, s: s + l, 0:2] = 0
    im[s: s + l, s + l: s + l + 3, 0:2] = 0
    im[s + l: s + l + 3, s: s + l, 0:2] = 0

    cv2.imwrite(fn, im)